client/allocrunner/taskrunner/task_runner.go

```diff
@@ -398,37 +436,67 @@ func (tr *TaskRunner) shouldRestart() (bool, time.Duration) {
 
 // runDriver runs the driver and waits for it to exit
 func (tr *TaskRunner) runDriver() error {
-	// Run prestart
-	ctx := driver.NewExecContext(tr.taskDir, tr.envBuilder.Build())
-	_, err := tr.driver.Prestart(ctx, tr.task)
-	if err != nil {
-		tr.logger.Error("driver pre-start failed", "error", err)
-		return err
+
+	taskConfig := drivers.NewTaskConfig(tr.task, tr.taskDir, tr.envBuilder.Build())
+	taskConfig.ID = tr.buildID()
+	taskConfig.StdoutPath = tr.logmonHookConfig.stdoutFifo
+	taskConfig.StderrPath = tr.logmonHookConfig.stderrFifo
```

Comment:
The constructor here starts to get pretty big. I could see adding a taskID arg, but this seems too messy:
```
func NewTaskConfig(taskID string, task *structs.Task, taskDir *allocdir.TaskDir, env *env.TaskEnv, stdoutPath, stderrPath string) *TaskConfig {...}
```
Another option could be to add a method to the task runner? 
```
func (tr *TaskRunner) buildTaskConfig() *drivers.TaskConfig{...}
```
I do sort of like this more because the ID generation could live here and not need to be its own func.